packages/unified-latex-util-arguments/libs/gobble-single-argument.ts

```diff
@@ -212,3 +210,97 @@ function partialStringMatch(node: Ast.Node, token: Ast.Node): boolean {
         node.content.length > token.content.length
     );
 }
+
+function cloneStringNode(node: Ast.String, content: string): Ast.String {
+    return Object.assign({}, node, { content });
+}
+
+/**
+ * Find the position of the open brace and the closing brace.
+ * Returns undefined if the brace isn't found.
+ * This may mutate `nodes`, if braces are not a kind of characters that are
+ * always parsed as a separate token
+ */
+function findBracePositions(nodes: Ast.Node[], startPos: number,
+    openMark?: string, closeMark?: string | Ast.Node): [number, number] | undefined {
+    const currNode = nodes[startPos];
+    let openMarkPos = startPos;
+    let closeMarkPos: number | null = startPos;
+    if (openMark) {
+        if (!match.anyString(currNode)) return;
```

Comment:
Is `whitespace` forbidden during a brace search? Not always, right?

Reply:
As I understand, `findBracePositions` is always (or should be) called after `gobbleWhitespaces` is called.
So `currNode` is always a non-whitespace node. That said, I realized that I should call `gobbleWhitespaces` between embellishment loops, fixed in https://github.com/siefkenj/unified-latex/pull/42/commits/052b4ff7c4ffbe8017ff0a780054133f6b77bf87
But this may become obsolete once the gobbleSingleArgument function is refactored to return single argument.